### DISCUSSION TITLE
Get a list of nodes that are not connected

### DISCUSSION THREAD

**Discussion by PixeledCode** [xyflow/xyflow#3077] [Q&A] [answered]
Hi, I was wondering if there is some utility available (I searched through docs, couldn't find anything) that will return the nodes that are not connected to any other node. Node without connected edges.

If there isn't, whats the most efficient way to implement this feature?
Thanks

**Comment by moklick** [comment, marked as answer]
you could filter all nodes that do not have edges:

```js
const unconnectedNodes = useStore(s => 
  s.getNodes().filter(node => !s.edges.some(edge => edge.source === node.id || edge.target === node.id))
);
```
